File: 1776-minimum-operations-to-reduce-x-to-zero/minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: list[int], x: int) -> int:
        n=len(nums)
        total=sum(nums)
        target=total-x
        start=end=0
        max_len=-1
        res=0
        if target < 0:
            return -1
        while end<n:
            res+=nums[end]
            while res>target :
                res-=nums[start]
                start+=1
            if res==target:
                max_len=max(max_len,end-start+1)
            end+=1
        if max_len==-1:
            return -1
        return n-max_len

File: 1776-minimum-operations-to-reduce-x-to-zero/minimum-operations-to-reduce-x-to-zero.java
class Solution {
    public int minOperations(int[] nums, int x) {
        int n = nums.length;
        int total = 0;
        for (int num : nums) {
            total += num;
        }
        
        int target = total - x;
        if (target < 0) {
            return -1;
        }
        
        int start = 0, end = 0, maxLen = -1;
        int res = 0;
        
        while (end < n) {
            res += nums[end];
            
            while (res > target && start <= end) {
                res -= nums[start];
                start++;
            }
            
            if (res == target) {
                maxLen = Math.max(maxLen, end - start + 1);
            }
            end++;
        }
        
        if (maxLen == -1) {
            return -1;
        }
        return n - maxLen;
    }
}